pkg/filters/proxies/grpcproxy/proxy.go

```diff
@@ -159,6 +212,45 @@ func (p *Proxy) reload() {
 			p.candidatePools = append(p.candidatePools, pool)
 		}
 	}
+
+	p.borrowTimeout, _ = time.ParseDuration(p.spec.BorrowTimeout)
+	p.timeout, _ = time.ParseDuration(p.spec.Timeout)
+	p.connectTimeout, _ = time.ParseDuration(p.spec.ConnectTimeout)
+
+	if p.pool == nil {
```

Comment:
this should always be true, so no need to check.

Reply:
when updating the proxy, p.pool is not nil.